pkg/services/dns/hosts_file.go

```diff
@@ -0,0 +1,94 @@
+package dns
+
+import (
+	"net"
+	"sync"
+
+	"github.com/areYouLazy/libhosty"
+	"github.com/fsnotify/fsnotify"
+	log "github.com/sirupsen/logrus"
+)
+
+type HostsFile interface {
+	LookupByHostname(name string) (net.IP, error)
+}
```

Comment:
As far as I can tell, only `hosts` is implemented this interface, I'd say the interface is not strictly required, this would work too:
```
type HostsFile struct {
	hostsReadLock sync.RWMutex
	hostsFilePath string
	hostsFile     *libhosty.HostsFile
}
```
If you prefer having an interface, fine with me.